### PR TITLE
Proposal for #1270

### PR DESCRIPTION
**Pull request by tejinderghatore**
This PR contain the changes for https://github.com/killbill/killbill/issues/1270

### CODE REVIEW

File: catalog/src/main/java/org/killbill/billing/catalog/CatalogUpdater.java [hunk 1]
```diff
@@ -56,6 +56,17 @@
...
+    
+    public static String INVALID_PLAN = "Plan doesn't exist or Plan is Invalid.";
+    
+    public static String INVALID_PRODUCT_NAME = "Product Name doesnt exist. Please provide valid Product Name.";
+    
+    public static String INVALID_PRICE = "Please check Amount and Currency. Amount should be greater than 0 and currency should be valid.";
+    
+    public static String BASE_PLAN_PRODUCTS_NOT_EMPTY = "Base Plan Products should not be empty.";
+    
+    public static String EXISTING_PRODUCTS_NOT_EMPTY = "Existing current product should not be empty.";
```
Comment (sbrossie) [inline catalog/src/main/java/org/killbill/billing/catalog/CatalogUpdater.java:68]: This error message is really not clear - and misleading. The issue here is that the client input contains a list of available products associated with the add-on, and when the code check if such products exist it fails, so basically the list provided contains reference to invalid products.

**Review by sbrossie — CHANGES_REQUESTED** [review]

File: catalog/src/main/java/org/killbill/billing/catalog/CatalogUpdater.java [hunk 2]
```diff
@@ -56,6 +56,17 @@
 public class CatalogUpdater {
 
     public static String DEFAULT_CATALOG_NAME = "DEFAULT";
+    
+    public static String INVALID_PLAN = "Plan doesn't exist or Plan is Invalid.";
```
[the remark this answers is not in the document]
Comment (tejinderghatore) [reply catalog/src/main/java/org/killbill/billing/catalog/CatalogUpdater.java]: I have changed to "Plan is invalid. Please check."

File: catalog/src/main/java/org/killbill/billing/catalog/CatalogUpdater.java [hunk 3]
```diff
@@ -56,6 +56,17 @@
...
+    
+    public static String INVALID_PRODUCT_NAME = "Product Name doesnt exist. Please provide valid Product Name.";
```
[the remark this answers is not in the document]
Comment (tejinderghatore) [reply catalog/src/main/java/org/killbill/billing/catalog/CatalogUpdater.java]: Fixed. Please check.

File: catalog/src/main/java/org/killbill/billing/catalog/CatalogUpdater.java [hunk 4]
```diff
@@ -56,6 +56,17 @@
...
+    
+    public static String INVALID_PRICE = "Please check Amount and Currency. Amount should be greater than 0 and currency should be valid.";
+    
+    public static String BASE_PLAN_PRODUCTS_NOT_EMPTY = "Base Plan Products should not be empty.";
```
[the remark this answers is not in the document]
Comment (tejinderghatore) [reply catalog/src/main/java/org/killbill/billing/catalog/CatalogUpdater.java]: Fixed. Please check now.

File: catalog/src/main/java/org/killbill/billing/catalog/CatalogUpdater.java (hunk 1 above)
Comment (tejinderghatore) [reply catalog/src/main/java/org/killbill/billing/catalog/CatalogUpdater.java]: I have the error description to "Available base products contain invalid product.Please check."

**Review by sbrossie — APPROVED** [review]
Happy to merge it as-is as it an improvement - I think we could have done even better:

Prior the change we had a poor description of the error but we had access to some of the `id` for the product/plan/... causing the issue. Now we have a better description but we lost the id. We could have created formatted messages that provide the best of both worlds.

**Pull request merged**